File: TravelAgencyFrontend/src/app/types/Travel/TravelResponse.ts
import {TransportType} from "./TransportType";
import {TravelState} from "./TravelState";
import {DayTripResponse} from "../DayTrip/DayTripResponse";
import {Pack} from "../Pack/Pack";
import {GuideResponse} from "../Guide/GuideResponse";
import {ClientResponse} from "../Client/ClientResponse";
import {CompanionResponse} from "../Companion/CompanionResponse";

export interface TravelResponse {
  id: string;
  title: string;
  destination: string;
  startDate: string; // ISO date string
  endDate: string; // ISO date string
  imagesUrl: string[]; // Array of image URLs
  transportType: TransportType;
  transportCompany: string;
  travelState: TravelState;
  travelGuide: GuideResponse;
  packs: Pack[];
  dayTrips: DayTripResponse[];
  clients: ClientResponse[];
  companions: CompanionResponse[];
  reservedSeat: number;
  maxSeat: number;
  basedPrice: number;
  shortDescription: string;
  longDescription: string;

}

File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/response/TravelResponse.java
package com.travelagency.travelagencyapplication.dto.response;

import com.travelagency.travelagencyapplication.collection.Client;
import com.travelagency.travelagencyapplication.collection.Companion;
import com.travelagency.travelagencyapplication.collection.User;
import com.travelagency.travelagencyapplication.dto.request.DayTripRequest;
import com.travelagency.travelagencyapplication.dto.request.PackRequest;
import com.travelagency.travelagencyapplication.enums.TransportType;
import com.travelagency.travelagencyapplication.enums.TravelState;
import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TravelResponse {

    private String id;
    private String title;
    private String destination;
    private LocalDate startDate;
    private LocalDate endDate;
    private Set<String> imagesUrl;

    private TransportType transportType;
    private String transportCompany;
    private TravelState travelState;

    private GuideResponse travelGuide;
    private Set<PackResponse> packs;
    private Set<DayTripResponse> dayTrips;

    private Set<ClientResponse> clients;

    private Set<CompanionResponse> companions;

    private Set<ClientResponse> canceledReservations;

    private Integer reservedSeat;
    private Integer maxSeat;
    private Double basedPrice;

    private String shortDescription;
    private String longDescription;
}
